browser/modules/DirectoryLinksProvider.jsm

```diff
@@ -720,20 +720,19 @@ let DirectoryLinksProvider = {
         // Only add data for non-empty tiles
         if (site) {
           // Remember which tiles data triggered the action
-          let link = site;
           let tilesIndex = tiles.length;
           if (triggeringSiteIndex == pos) {
             actionIndex = tilesIndex;
           }
 
           // Make the payload in a way so keys can be excluded when stringified
-          let id = link.directoryId;
+          let id = site.directoryId;
           tiles.push({
             id: id || site.enhancedId,
-            pin: link.pinState ? 1 : undefined,
+            pin: site.pinState ? 1 : undefined,
```

Comment:
I'm concerned that sometimes we set things to `undefined` and other times to `null`. What's the rationale for using "null" in some places but "undefined" in others?